### PR TITLE
NF-230 does importer have eori

### PR DESCRIPTION
**Pull request by rorticus0502**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala [hunk 1]
```diff
@@ -71,6 +72,8 @@ protected trait Conditions {
 
   protected val isRepresentative: UserAnswers => Boolean = _.isRepresentative
 
+  protected val importerHasEori: UserAnswers => Boolean = _.doesImporterHaveEori
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:75]: Don't need this method ATM.  But you can leave it

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala [hunk 2]
```diff
@@ -40,13 +40,15 @@ class NavigatorSpec extends UnitSpec with Injector with TestData {
     val previousPage = back(AddressPage, _)
 
     "going forward" should {
-      "go to bank details page" in {
+      "dont go to importer has EORI page for importer" in {
```
Comment (SteveSugden) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala:43]: "go to bank details page if claimant is importer" ?

Comment (SteveSugden) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala:43]: But these tests will change when I merge my EORI branch in.

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala [hunk 3]
```diff
@@ -56,6 +58,27 @@ class NavigatorSpec extends UnitSpec with Injector with TestData {
     }
   }
 
+  "Navigator from importer eori page" when {
```
Comment (SteveSugden) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala:61]: I'm not sure about adding even more "navigate from page x" tests.  We're already having to fix tests when pages are added.   What do you think @crosslandwa ?

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala [hunk 4]
```diff
@@ -40,13 +40,15 @@ class NavigatorSpec extends UnitSpec with Injector with TestData {
     val previousPage = back(AddressPage, _)
 
     "going forward" should {
-      "go to bank details page" in {
+      "dont go to importer has EORI page for importer" in {
         nextPage(
           answers().copy(representationType = Some(RepresentationType.Importer))
         ) mustBe routes.BankDetailsController.onPageLoad()
+      }
+      "do go to importer has EORI page for representative" in {
```
Comment (SteveSugden) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala:48]: "go to does importer have EORI page if claimant is representative" ?

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala (hunk 3 above)
Comment (crosslandwa) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala:61]: I think adding more tests like this:
 - doesn't make page order any easier to infer
 - makes page order harder to change
 
And therefore does not add value


For a developer working with this code in isolation, I think inferring the page order from the `Seq` within the `Navigator` implementation is straightforward - if the implementation were more complex, I'd see more value in tests that described the page order...

Also, considering the service "as a whole", the overall page order of the journey is documented/tested in the acceptance tests (in the smoke tests)

File: conf/app.routes [hunk 5]
```diff
@@ -24,6 +24,9 @@ POST        /contact-details                    uk.gov.hmrc.nationalimportdutyad
 GET         /your-address                       uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.AddressController.onPageLoad()
 POST        /your-address                       uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.AddressController.onSubmit()
 
+GET         /eori-number                       uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.ImporterHasEoriController.onPageLoad()
```
Comment (crosslandwa) [inline conf/app.routes:27]: CLARIFICATION Did this URL come from the content designer? Are we sure it is ok _without_ specifying "importer" in the URL (e.g. `/importers-eori-number`)?

Comment (SteveSugden) [reply conf/app.routes:27]: It is in the design doc, but that is a good point.  And maybe it should be in the form of a question.
The page I've done which captures the EORI is `/enter-eori-number`

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala (hunk 1 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:75]: I will leave it then. I agree unused code should not be there but in this case we were kind of doing half a story each and I wanted to include the conditional nextpage bit, for my own understanding really.

File: conf/app.routes (hunk 5 above)
Comment (rorticus0502) [reply conf/app.routes:27]: The url was specified in the document, linked to from the jira ticket.

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala (hunk 2 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala]: changed the name as suggested - much clearer

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala (hunk 3 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala]: additional test removed. Again it was more for my benefit than anything else.

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala (hunk 4 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/NavigatorSpec.scala]: much clearer - name has been changed

**Review by SteveSugden — DISMISSED** [review]
I'm happy.

**Review by SteveSugden — DISMISSED** [review]

**Review by SteveSugden — APPROVED** [review]

**Pull request merged**